src/pudl/transform/ferc1.py

```diff
@@ -2976,6 +2976,37 @@ def remove_nones_in_list(list_of_things):
             tbl_meta.loc[xbrl_factoid, "calculations"] = json.dumps(calc_to_update)
         return tbl_meta.reset_index()
 
+    def process_xbrl_metadata_calculations(
+        self,
+    ) -> pd.DataFrame:
+        """Convert xbrl metadata calculations into a table of calculation components."""
+        metadata = self.xbrl_metadata
+        if all(metadata.calculations == "[]"):
+            logger.info(f"{self.table_id.value}: No calculations.")
+            return pd.DataFrame()
+
+        metadata.calculations = metadata.calculations.apply(json.loads)
+        # reset the index post calc explosion so we can merge on index later
+        metadata = metadata.explode("calculations").reset_index(drop=True)
+        calc_comps = (
+            pd.json_normalize(metadata.calculations)
+            .explode("source_tables")
+            .rename(
+                columns={
+                    "name": "xbrl_factoid_calculation_component",
```

Comment:
Those names read better, but I think these are the copies of the PK columns that are associated with the calculation components in the association table, so just having a suffix to differentiate them from the "main" versions of the same columns makes sense.  But I think it could just be `_calc` and that would be fine.